### ISSUE TITLE
Could not compile instructor in livebook

### ISSUE THREAD

**Issue by Munksgaard** [thmsmlr/instructor_ex#25]
When trying to run [the quickstart example](https://hexdocs.pm/instructor/quickstart.html) in my local livebook, I get this error:

```
Could not compile :instructor, no "mix.exs", "rebar.config" or "Makefile" (pass :compile as an option to customize compilation, set it to "false" to do nothing)
Unchecked dependencies for environment dev:
* instructor (/home)
  could not find an app file at "_build/dev/lib/instructor/ebin/instructor.app". This may happen if the dependency was not yet compiled or the dependency indeed has no app file (then you can pass app: false as option)
```

**Comment by Munksgaard** [comment]
Looks like the problem is the `path` specification.

**Comment by Munksgaard** [comment]
Thank you, that works. I realize that changing this in the repo would mean having to update every time there's a new release, but having it like it currently is kind of defeats the purpose, imho. How do other libraries handle this?

**Comment by TwistingTwists** [comment]
You can use git source. 

`{:foobar, git: "https://github.com/elixir-lang/foobar.git", tag: "0.1"}
`

**Comment by TwistingTwists** [comment]
You can put branch name. 
`{:foobar, git: "https://github.com/elixir-lang/foobar.git", branch: "master"}`

**Comment by Munksgaard** [comment]
Yeah, either could work. I leave it up to you to choose whether to change anything or just close this issue :-)

**Comment by thmsmlr** [comment]
Yeah.. no solution is great. Using :path like i am currently breaks when you try to import the livebook. Version pegging is tedious when you're developing new examples pre-release. Git suffers from the same issue, but may be the best option since at least you can develop docs against unreleased code, pre-release?

**Comment by dewalor** [comment]
> You can use git source.
> 
> `{:foobar, git: "https://github.com/elixir-lang/foobar.git"} `

like this:
`{:instructor, git: "https://github.com/thmsmlr/instructor_ex.git", branch: "main"}`

**Comment by dewalor** [comment]
defmodule President do
  use Ecto.Schema

  @primary_key false
  embedded_schema do
    field(:first_name, :string)
    field(:last_name, :string)
    field(:entered_office_date, :date)
  end
end

Instructor.chat_completion(
  mode: :json,
  model: "mistral:7b-instruct-q6_K",
  response_model: President,
  messages: [
    %{role: "user", content: "Who was the first president of the United States?"}
  ]
)

**Comment by dewalor** [comment]
Instructor.chat_completion(
  model: "nous-hermes2-mixtral:8x7b-dpo-q6_K",
  mode: :json,
  stream: true,
  response_model: {:array, President},
  messages: [
    %{role: "user", content: "Who are the first three presidents"}
  ]
)
|> Stream.each(fn {:ok, x} -> IO.inspect(x) end)
|> Stream.run()

**Comment by thmsmlr** [comment]
> > You can use git source.
> > `{:foobar, git: "https://github.com/elixir-lang/foobar.git"} `
> 
> like this: `{:instructor, git: "https://github.com/thmsmlr/instructor_ex.git", branch: "main"}`

I don't hate this, but it still doesn't solve the local development issue. I often am updating the documentation while I'm working on a local commit.
